### PR TITLE
Resolves#16 -Refactor code to adhere to the rules in the description

### PR DESCRIPTION
**Pull request by Farhanisto**
Resolves [#16](https://github.com/AndelaOSP/andela-societies-frontend/issues/16)

Description:
Refactor code to adhere to the rules in the description

Fix:
placed Proptypes on top, placed state inside the constructor and used arrow function instead of binding functions

### CODE REVIEW

File: src/containers/MasonryLayout.jsx [hunk 1]
```diff
@@ -8,6 +8,16 @@ import PropTypes from 'prop-types';
  * @extends React.component
  */
 class MasonryLayout extends Component {
+  static defaultProps = {
```
Comment (john555) [inline src/containers/MasonryLayout.jsx]: You omitted comments. We need the comments @Farhanisto

File: src/components/activities/ActivityCard.jsx [hunk 2]
```diff
@@ -7,6 +7,13 @@ import PropType from 'prop-types';
  * @extends React.Component
 */
 class ActivityCard extends Component {
+  static propTypes = {
```
Comment (john555) [inline src/components/activities/ActivityCard.jsx]: You omitted comments

File: src/components/activities/ActivityCard.jsx [hunk 3]
```diff
@@ -7,6 +7,13 @@ import PropType from 'prop-types';
...
+    category: PropType.string.isRequired,
+    date: PropType.string.isRequired,
+    description: PropType.string.isRequired,
+    points: PropType.string.isRequired,
+    status: PropType.string.isRequired,
+  };
   constructor(props) {
```
Comment (joykare) [inline src/components/activities/ActivityCard.jsx]: For this component we do not yet need the constructor coz there is no state we need to initialize

File: src/components/header/Header.jsx [hunk 4]
```diff
@@ -11,15 +11,16 @@ import Breadcrumb from './Breadcrumb';
...
  */
 export default class Header extends Component {
-  state = {
-    ...this.initialState,
-  };
-
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...this.initialState,
```
Comment (joykare) [inline src/components/header/Header.jsx]: why not just put the contents of `initialState` in `this.state`

File: src/components/activities/ActivityCard.jsx (hunk 2 above)
Comment (joykare) [reply src/components/activities/ActivityCard.jsx]: yeah just copy<>paste the comments @john555 had already added

**Comment by joykare** [comment]
You also haven't changed any event handling instances. I'm sure there are a number. Coz we saw them the day we went through the code.

File: src/components/header/Header.jsx (hunk 4 above)
Comment (Farhanisto) [reply src/components/header/Header.jsx]: Ok @joykare  will do

File: src/components/activities/ActivityCard.jsx (hunk 3 above)
Comment (Farhanisto) [reply src/components/activities/ActivityCard.jsx]: Thanks for the pointer @joykare  remove ill remove this constructor

File: src/components/activities/ActivityCard.jsx [hunk 5]
```diff
@@ -7,16 +7,28 @@ import PropType from 'prop-types';
...
+    description: PropType.string.isRequired,
+    points: PropType.string.isRequired,
+    status: PropType.string.isRequired,
+  };
+  statuses = ['pending', 'expired', 'approved', 'default'];
   /**
    * @summary Renders the status indicator on the ActivityCard
    */
-  renderStatus() {
+  renderStatus= () => {
```
Comment (john555) [inline src/components/activities/ActivityCard.jsx]: Add space before equal sign.

File: src/containers/SignIn.jsx [hunk 6]
```diff
@@ -59,7 +59,9 @@ class SignIn extends Component {
                 </div>
                 <div className='feature__info'>
                   <h2 className='feature__name'>Verify Activities</h2>
-                  <p className='feature__description'>Society secretaries and the success department can verify activities.</p>
+                  <p className='feature__description'>
+                  Society secretaries and the success department can verify activities.
```
Comment (mkiterian) [inline src/containers/SignIn.jsx]: This should be indented

File: src/components/activities/ActivityCard.jsx (hunk 5 above)
Comment (georgreen) [reply src/components/activities/ActivityCard.jsx]: @john555 @joykare @Farhanisto Addressing style rules on PR reviews is a good practise, on the other hand though it makes PR reviews noisy and distracting from the global objective i.e to make sure code is correct, secure and flexible. I have an opinion, that using linters and CI to make sure such minor details don't make it to this level will be much better. I'm suggesting we adopt this in our work flow, any thoughts on this @team?

File: src/components/svgIcons/societyIcons/Istelle.jsx [hunk 7]
```diff
@@ -1,15 +1,40 @@
 import React from 'react';
 
+const svgPointA = 'M239,317.54c4.92,0.32,9.84.63,14.76,1q8.71,0.62,17.42,1.3c3.31,0.26,6.63.47,9.93,0.88,5.52,0.68,9,' +
```
Comment (john555) [inline src/components/svgIcons/societyIcons/Istelle.jsx]: This is too much work for a minor thing. You could disable the line-length rule for this component and others like it. Nobody ever reads SVG anyway.

Comment (Farhanisto) [reply src/components/svgIcons/societyIcons/Istelle.jsx]: Now the work is done.

**Comment by john555** [comment]
LGTM

**Comment by Farhanisto** [comment]
@joykare I have completed on the requested changes could you kindly check.

**Pull request merged**